rush.json

```diff
@@ -1,20 +1,300 @@
+   *
+   *   common/config/rush/variants/<variant_name>
+   *
+   * The variant folder can contain an alternate common-versions.json file.  Its "preferredVersions" field can be used
+   * to select older versions of dependencies (within a loose SemVer range specified in your package.json files).
+   * To install a variant, run "rush install --variant <variant_name>".
+   *
+   * For more details and instructions, see this article:  https://rushjs.io/pages/advanced/installation_variants/
+   */
+  "variants": [
+    // {
+    //   /**
+    //    * The folder name for this variant.
+    //    */
+    //   "variantName": "old-sdk",
+    //
+    //   /**
+    //    * An informative description
+    //    */
+    //   "description": "Build this repo using the previous release of the SDK"
+    // }
+  ],
+
+  /**
+   * Rush can collect anonymous telemetry about everyday developer activity such as
+   * success/failure of installs, builds, and other operations.  You can use this to identify
+   * problems with your toolchain or Rush itself.  THIS TELEMETRY IS NOT SHARED WITH MICROSOFT.
+   * It is written into JSON files in the common/temp folder.  It's up to you to write scripts
+   * that read these JSON files and do something with them.  These scripts are typically registered
+   * in the "eventHooks" section.
+   */
+  // "telemetryEnabled": false,
+
+  /**
+   * Allows creation of hotfix changes. This feature is experimental so it is disabled by default.
+   */
+  // "hotfixChangeEnabled": false,
+
+  /**
+   * (Required) This is the inventory of projects to be managed by Rush.
+   *
+   * Rush does not automatically scan for projects using wildcards, for a few reasons:
+   * 1. Depth-first scans are expensive, particularly when tools need to repeatedly collect the list.
+   * 2. On a caching CI machine, scans can accidentally pick up files left behind from a previous build.
+   * 3. It's useful to have a centralized inventory of all projects and their important metadata.
+   */
   "projects": [
     {
+      /**
+       * The NPM package name of the project (must match package.json)
+       */
       "packageName": "office-ui-fabric-react",
+      /**
+       * The path to the project folder, relative to the rush.json config file.
+       */
       "projectFolder": "packages/office-ui-fabric-react",
+      /**
+       * An optional version policy associated with the project.  Version policies are defined
+       * in "version-policies.json" file.  See the "rush publish" documentation for more info.
+       * NOTE: "versionPolicyName" and "shouldPublish" are alternatives; you cannot specify them both.
+       */
       "versionPolicyName": "lockedMajor",
+      /**
+       * A flag indicating that changes to this project will be published to npm, which affects
+       * the Rush change and publish workflows. The default value is false.
+       * NOTE: "versionPolicyName" and "shouldPublish" are alternatives; you cannot specify them both.
```

Comment:
The note here says we can't have both versionPolicyName and shouldPublish, but that seems to be the case.
Any particular problems this might cause?

Reply:
shouldPublish is ignored with a version Policy so in that sense it is relatively harmless... The problem mostly lie with the shouldPublish=false and versionPolicyVersion="lockedMajor" combination - in which case it still publishes despite what you think you told Rush to do. 

But great catch. I really wish Rush would complain about the bad combination above.

---
In reply to: [236154610](https://github.com/OfficeDev/office-ui-fabric-react/pull/7186#discussion_r236154610) [](ancestors = 236154610)

Reply:
Opened [an issue](https://github.com/Microsoft/web-build-tools/issues/958) about this. :)